tx/transaction.go

```diff
@@ -126,24 +157,20 @@ func (t *Transaction) UnprovedWork() (w *big.Int) {
 	if err != nil {
 		return &big.Int{}
 	}
-	return t.EvaluateWork(origin)(t.body.Nonce)
+	return t.EvaluateWork(origin)(t.body.nonce())
 }
 
 // EvaluateWork try to compute work when tx origin assumed.
 func (t *Transaction) EvaluateWork(origin thor.Address) func(nonce uint64) *big.Int {
-	hashWithoutNonce := thor.Blake2bFn(func(w io.Writer) {
-		rlp.Encode(w, []interface{}{
-			t.body.ChainTag,
-			t.body.BlockRef,
-			t.body.Expiration,
-			t.body.Clauses,
-			t.body.GasPriceCoef,
-			t.body.Gas,
-			t.body.DependsOn,
-			&t.body.Reserved,
-			origin,
-		})
-	})
+	var hashWithoutNonce *thor.Bytes32
```

Comment:
I wonder if the panic could be avoided by deferring the hashing  to something like: `hashWithoutNonce := t.body.hashWithoutNonce(origin)` ?
If the legacy and dynfee tx had their own impl ?

Reply:
yeah, makes sense